### ISSUE TITLE
False positive reportPossiblyUnboundVariable with dynamic or lazy imports

### ISSUE THREAD

**Issue by ssbarnea** [microsoft/pyright#9494]
**Describe the bug**

Apparently pyright is not able to detect any conditional imports. This is a big deal as there is a lot of production code that needs to make use of these. For example mypy has no problem dealing with these.

```python
if "run" not in globals():
    from subprocess import run

run("echo 123")
```

```
 error: "run" is possibly unbound (reportPossiblyUnboundVariable)
```

pyright 1.1.389

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

In your code sample above, the symbol `run` is not defined prior to the `if` statement. Perhaps your sample is incomplete? How would `run` receive a value prior to the import statement otherwise?

**Comment by ssbarnea** [comment]
That code is valid python code. The sample is complete. I do think that the assumption that python imports happen only at the beginning of the file is the issue here.

Basically any code that imports something if is not already imported will fail `reportPossiblyUnboundVariable`. It name is misleading because the execution will never reach the `run` line if the import fails, it cannot really be unbound at that line.
